src/DFG.cpp

```diff
@@ -8,1798 +8,1971 @@
+             ((high_dvfs_dfg_nodes + num_tiles_in_island - 1) / num_tiles_in_island)) -
+     (t_II * num_tiles_in_island *
+             ((mid_dvfs_dfg_nodes + num_tiles_in_island - 1) / num_tiles_in_island))
+         ) / 4;
+   cout << "[debug] unused_high_dvfs_cgra_tiles_across_II: " << unused_high_dvfs_cgra_tiles_across_II << endl;
+   cout << "[debug] unused_mid_dvfs_cgra_tiles_across_II: " << unused_mid_dvfs_cgra_tiles_across_II << endl;
+   cout << "[debug] unused_low_dvfs_cgra_tiles_across_II: " << unused_low_dvfs_cgra_tiles_across_II << endl;
+ 
+   int unlabeled_dfg_nodes = 0;
+   for (auto node : nodes) {
+     if (assigned_dvfs_nodes.count(node) == 0) {
+       unlabeled_dfg_nodes += 1;
+     }
+   }
+   if (unlabeled_dfg_nodes > unused_low_dvfs_cgra_tiles_across_II) {
+     int min_reserved_low_dvfs_tiles_across_II = unused_low_dvfs_cgra_tiles_across_II / 4.5;
+     int num_low_dvfs_dfg_nodes = 0;
+     for (auto node : nodes) {
+       if (assigned_dvfs_nodes.count(node) == 0) {
+         node->setDVFSLatencyMultiple(4);
+         assigned_dvfs_nodes.insert(node);
+         num_low_dvfs_dfg_nodes += 1;
+         if (num_low_dvfs_dfg_nodes >= min_reserved_low_dvfs_tiles_across_II) {
+           unused_low_dvfs_cgra_tiles_across_II -= num_low_dvfs_dfg_nodes;
+           break;
+         }
+       }
+     }
+   } else {
+     for (auto node : nodes) {
+       if (assigned_dvfs_nodes.count(node) == 0) {
+         node->setDVFSLatencyMultiple(4);
+         assigned_dvfs_nodes.insert(node);
+       }
+     }
+     unused_low_dvfs_cgra_tiles_across_II -= unlabeled_dfg_nodes;
+   }
+ 
+   for (auto node : nodes) {
+     if (assigned_dvfs_nodes.count(node) == 0) {
+       if (unused_high_dvfs_cgra_tiles_across_II > 0) {
+         // High DVFS islands have the highest priority as we don't want to
+   // waste it.
+         node->setDVFSLatencyMultiple(1);
+         assigned_dvfs_nodes.insert(node);
+         unused_high_dvfs_cgra_tiles_across_II -= 1;
+   unused_mid_dvfs_cgra_tiles_across_II -= 1;
+   unused_low_dvfs_cgra_tiles_across_II -= 1;
+       } else if (unused_mid_dvfs_cgra_tiles_across_II > 0) {
+         // Then try to allocate the DFG node into the mid DVFS island if the
+   // high DVFS islands are used up.
+         node->setDVFSLatencyMultiple(2);
+         assigned_dvfs_nodes.insert(node);
+         unused_high_dvfs_cgra_tiles_across_II -= 2;
+         unused_mid_dvfs_cgra_tiles_across_II -= 1;
+         unused_low_dvfs_cgra_tiles_across_II -= 1;
+       } else if (unused_low_dvfs_cgra_tiles_across_II > 0) {
+         // Low DVFS islands have the lowest priority.
+         node->setDVFSLatencyMultiple(4);
+         assigned_dvfs_nodes.insert(node);
+         unused_high_dvfs_cgra_tiles_across_II -= 4;
+         unused_mid_dvfs_cgra_tiles_across_II -= 2;
+         unused_low_dvfs_cgra_tiles_across_II -= 1;
+       } else {
+   // If all the islands assuming the optimal II are used up, label
```

Comment:
Messed up here. I believe it is the IDE's issue, the "tab" and "space" are not interpreted as expected.